src/mutation.h

```diff
@@ -107,6 +107,52 @@ struct reflex_activation_data {
     void deserialize( const JsonObject &jo );
 };
 
+struct trait_and_var {
+    trait_id trait;
+    std::string variant;
+
+    trait_and_var() = default;
+    trait_and_var( const trait_id &t, const std::string &v ) : trait( t ), variant( v ) {}
+
+
+    void deserialize( const JsonValue &jv );
+    void serialize( JsonOut &jsout ) const;
+
+    std::string name() const;
+    std::string desc() const;
+
+    bool operator==( const trait_and_var &other ) const {
+        return trait == other.trait && variant == other.variant;
+    }
+};
+
+struct trait_replacement {
+    cata::optional<proficiency_id> prof;
+    cata::optional<trait_and_var> trait;
+    bool error = false;
+};
+
+struct mutation_variant {
+    std::string id;
```

Comment:
If it's an id, it should probably be a `string_id` instead of some arbitrary string.

Reply:
I didn't use a proper typed id because these aren't unique - what `string_id<mutation_variant>( "blue" )` refers to depends on the trait it's associated with. I'd be willing to, but I thought it was a little more clear this way.